pkg/local/diff.go

```diff
@@ -188,6 +186,59 @@ func Undo(c Changes) Changes {
 	return reversed
 }
 
+// WalkS3 - walk like an Egyptian... ¯\_(ツ)¯\_
+// This walker function simulates the way object listing is performed by S3. Contrary to how a standard FS walk function behaves, S3
+// does not take into consideration the directory hierarchy. Instead, object paths include the entire path relative to the root and as a result
+// the directory or "path separator" is also taken into account when providing the listing in a lexicographical order.
+func WalkS3(root string, callbackFunc func(p string, info fs.FileInfo, err error) error) error {
+	var dirQueue []string
+	err := filepath.Walk(root, func(p string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if p == root {
+			return nil
+		}
+		if info.IsDir() {
+			// TODO: We don't return dir results for the listing, how will this effect directory markers, and can we even support directory markers?
```

Comment:
This is a potentially severe product limitation in `lakectl local`:
1. Run Spark directly on lakeFS.
1. Get that branch into `lakectl local`.
1. Commit the branch "unchanged".

Won't we see (many) changes?

Reply:
This comment is not a new behavior. We didn't support directory markers also before. It came up while I was writing the fix and I thought it was worth mentioning

Reply:
Can we open an issue for this?

Reply:
https://github.com/treeverse/lakeFS/issues/7878